diff -ur rustc-1.28.0-src.old/src/bootstrap/dist.rs rustc-1.28.0-src/src/bootstrap/dist.rs
--- rustc-1.28.0-src.old/src/bootstrap/dist.rs	2018-08-04 21:19:13.304292468 +0300
+++ rustc-1.28.0-src/src/bootstrap/dist.rs	2018-08-05 05:52:12.731588011 +0300
@@ -482,8 +482,9 @@

             // Copy over the codegen backends
             let backends_src = builder.sysroot_codegen_backends(compiler);
-            let backends_rel = backends_src.strip_prefix(&src).unwrap();
-            let backends_dst = image.join(&backends_rel);
+            let libdir_relative = builder.build.config.libdir_relative().unwrap();
+            let backends_rel = backends_src.strip_prefix(&src).unwrap().strip_prefix(&libdir_relative).unwrap();
+            let backends_dst = image.join("lib").join(&backends_rel);
             t!(fs::create_dir_all(&backends_dst));
             builder.cp_r(&backends_src, &backends_dst);

