diff -up pip/src/pip/_internal/utils/temp_dir.py.old pip/src/pip/_internal/utils/temp_dir.py
--- pip/src/pip/_internal/utils/temp_dir.py.old	2021-07-08 21:32:25.841933736 +0300
+++ pip/src/pip/_internal/utils/temp_dir.py	2021-07-08 21:32:38.254938272 +0300
@@ -2,7 +2,7 @@ import errno
 import itertools
 import logging
 import os.path
-import tempfile
+from . import tempfile
 from contextlib import contextmanager
 
 from pip._vendor.contextlib2 import ExitStack
