From 714f41924b7a91623414487750bee8e85da47467 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Javier=20Jard=C3=B3n?= <jjardon@gnome.org>
Date: Sun, 15 Jul 2018 16:37:23 +0100
Subject: [PATCH] mozilla: Make it possible to overwrite python version

---
 mozilla/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mozilla/Makefile b/mozilla/Makefile
index 6f46118..ad9a714 100644
--- a/mozilla/Makefile
+++ b/mozilla/Makefile
@@ -2,8 +2,11 @@
 # Makefile
 #
 
+SHELL := env PYTHON=$(PYTHON) $(SHELL)
+PYTHON ?= "python2"
+
 all:
-	python certdata2pem.py
+	$(PYTHON) certdata2pem.py
 
 clean:
 	-rm -f *.crt
-- 
2.18.0

