diff --git a/config.c b/config.c
index b0dd721..c7f49e3 100644
--- a/config.c
+++ b/config.c
@@ -362,7 +362,7 @@ static struct config opts[] = {
 		.name = "genext2fs",
 		.opt = CFG_STR("genext2fs", NULL, CFGF_NONE),
 		.env = "GENIMAGE_GENEXT2FS",
-		.def = "genext2fs",
+		.def = "mke2fs",
 	}, {
 		.name = "genisoimage",
 		.opt = CFG_STR("genisoimage", NULL, CFGF_NONE),
diff --git a/image-ext2.c b/image-ext2.c
index 8ce76be..4c1313e 100644
--- a/image-ext2.c
+++ b/image-ext2.c
@@ -39,9 +39,10 @@ static int ext2_generate_genext2fs(struct image *image)
 	const char *extraargs = cfg_getstr(image->imagesec, "extraargs");
 	const char *label = cfg_getstr(image->imagesec, "label");
 
-	ret = systemp(image, "%s -d '%s' --size-in-blocks=%lld -i 16384 '%s' %s",
+	ret = systemp(image, "%s -d '%s' -i 16384 '%s' %lld %s",
 			get_opt("genext2fs"),
-			mountpath(image), image->size / 1024, imageoutfile(image),
+			mountpath(image), imageoutfile(image),
+                        image->size / 1024,
 			extraargs);
 
 	if (ret)
