diff -u -r libseccomp.old/tools/Makefile.am libseccomp/tools/Makefile.am
--- libseccomp.old/tools/Makefile.am	2019-02-20 21:11:15.490400781 +0100
+++ libseccomp/tools/Makefile.am	2019-02-20 21:11:40.878224899 +0100
@@ -35,8 +35,6 @@
 scmp_bpf_sim_SOURCES = scmp_bpf_sim.c bpf.h util.h
 
 scmp_sys_resolver_LDADD = ../src/libseccomp.la
-scmp_sys_resolver_LDFLAGS = -static
 scmp_arch_detect_LDADD = ../src/libseccomp.la
-scmp_arch_detect_LDFLAGS = -static
 scmp_bpf_disasm_LDADD = util.la
 scmp_bpf_sim_LDADD = util.la
