diff --git a/eos-updater/meson.build b/eos-updater/meson.build
index 825a6ec..dd4d5a1 100644
--- a/eos-updater/meson.build
+++ b/eos-updater/meson.build
@@ -40,7 +40,6 @@ libeos_updater_dbus = static_library('eos-updater-dbus',
   dependencies: libeos_updater_dbus_deps,
   include_directories: root_inc,
   install: false,
-  version: meson.project_version(),
 )
 libeos_updater_dbus_dep = declare_dependency(
   link_with: libeos_updater_dbus,
diff --git a/libeos-update-server/meson.build b/libeos-update-server/meson.build
index a11b619..517ea23 100644
--- a/libeos-update-server/meson.build
+++ b/libeos-update-server/meson.build
@@ -40,7 +40,6 @@ libeos_update_server = static_library('eos-update-server-' + eus_api_version,
   dependencies: libeos_update_server_deps,
   include_directories: root_inc,
   install: false,
-  version: meson.project_version(),
 )
 libeos_update_server_dep = declare_dependency(
   link_with: libeos_update_server,
diff --git a/test-common/meson.build b/test-common/meson.build
index 9e47567..95e5c49 100644
--- a/test-common/meson.build
+++ b/test-common/meson.build
@@ -58,7 +58,6 @@ libeos_updater_test_common = static_library('eos-updater-test-common',
   dependencies: libeos_updater_test_common_deps,
   include_directories: root_inc,
   install: false,
-  version: meson.project_version(),
 )
 libeos_updater_test_common_dep = declare_dependency(
   link_with: libeos_updater_test_common,
