diff -ur libunwind.old/configure.ac libunwind/configure.ac
--- libunwind.old/configure.ac	2019-05-14 13:13:10.154495351 +0200
+++ libunwind/configure.ac	2019-05-14 13:13:20.730393914 +0200
@@ -200,12 +200,6 @@
 AM_CONDITIONAL(USE_DWARF, [test x$use_dwarf = xyes])
 AC_MSG_RESULT([$use_dwarf])
 
-if test x$target_arch = xppc64; then
-        libdir='${exec_prefix}/lib64'
-        AC_MSG_NOTICE([PowerPC64 detected, lib will be installed ${libdir}]);
-        AC_SUBST([libdir])
-fi
-
 AC_MSG_CHECKING([whether to restrict build to remote support])
 if test x$target_arch != x$host_arch; then
   CPPFLAGS="${CPPFLAGS} -DUNW_REMOTE_ONLY"
