diff -ur gbm.old/src/gbm/main/backend.c gbm/src/gbm/main/backend.c
--- gbm.old/src/gbm/main/backend.c	2021-08-10 09:12:04.696844111 +0200
+++ gbm/src/gbm/main/backend.c	2021-08-10 09:13:51.698031448 +0200
@@ -51,7 +51,7 @@
 };
 
 static const struct gbm_backend_desc builtin_backends[] = {
-   { "dri", &gbm_dri_backend },
+/*   { "dri", &gbm_dri_backend },*/
 };
 
 #define BACKEND_LIB_SUFFIX "_gbm"
