diff --git a/src/broadcom/cle/meson.build b/src/broadcom/cle/meson.build
index 50031cc7137..ede3b174fe5 100644
--- a/src/broadcom/cle/meson.build
+++ b/src/broadcom/cle/meson.build
@@ -54,7 +54,7 @@ v3d_xml_h = custom_target(
 )
 
 libbroadcom_cle = static_library(
-  ['broadcom_cle', v3d_xml_h],
+  ['broadcom_cle', v3d_xml_h, v3d_xml_pack],
   'v3d_decoder.c',
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_broadcom],
   c_args : [no_override_init_args],
