diff -ur gcc.old/libstdc++-v3/src/c++17/Makefile.am gcc/libstdc++-v3/src/c++17/Makefile.am
--- gcc.old/libstdc++-v3/src/c++17/Makefile.am	2021-05-06 11:12:23.447901790 +0200
+++ gcc/libstdc++-v3/src/c++17/Makefile.am	2021-05-06 13:20:43.531772731 +0200
@@ -79,6 +79,7 @@
 # OPTIMIZE_CXXFLAGS on the compile line so that -O2 can be overridden
 # as the occasion calls for it.
 AM_CXXFLAGS = \
+	-nostdinc++ \
 	-std=gnu++17 \
 	$(glibcxx_lt_pic_flag) $(glibcxx_compiler_shared_flag) \
 	$(XTEMPLATE_FLAGS) $(VTV_CXXFLAGS) \
