diff --git a/Makefile b/Makefile
index a0b12e7de..5207c690e 100644
--- a/Makefile
+++ b/Makefile
@@ -430,7 +430,7 @@ pkg/api/swagger.yaml: .gopathok
 $(MANPAGES): %: %.md .install.md2man docdir
 	@sed -e 's/\((podman[^)]*\.md)\)//g' -e 's/\[\(podman[^]]*\)\]/\1/g' \
 		-e 's;<\(/\)\?\(a\|a\s\+[^>]*\|sup\)>;;g' $<  | \
-	$(GOMD2MAN) -in /dev/stdin -out $(subst source/markdown,build/man,$@)
+	$(GOMD2MAN) -in /proc/self/fd/0 -out $(subst source/markdown,build/man,$@)
 
 .PHONY: docdir
 docdir:
