diff --git a/mesonbuild/mesonlib/universal.py b/mesonbuild/mesonlib/universal.py
index 1b5b86c20..e69c977d4 100644
--- a/mesonbuild/mesonlib/universal.py
+++ b/mesonbuild/mesonlib/universal.py
@@ -854,7 +854,7 @@ def default_libdir() -> str:
             pass
     if is_freebsd() or is_irix():
         return 'lib'
-    if os.path.isdir('/usr/lib64') and not os.path.islink('/usr/lib64'):
+    if os.path.isdir('/usr/lib64') and not os.path.islink('/usr/lib64') and False:
         return 'lib64'
     return 'lib'
 
