Index: llvm-toolchain-snapshot_13~++20210325124256+06411edb9fca/clang/lib/Driver/ToolChains/Linux.cpp
===================================================================
--- llvm-toolchain-snapshot_13~++20210325124256+06411edb9fca.orig/clang/lib/Driver/ToolChains/Linux.cpp
+++ llvm-toolchain-snapshot_13~++20210325124256+06411edb9fca/clang/lib/Driver/ToolChains/Linux.cpp
@@ -115,6 +115,10 @@ std::string Linux::getMultiarchTriple(co
       return "mips64el-linux-gnu";
     break;
   }
+  case llvm::Triple::riscv64:
+    if (D.getVFS().exists(SysRoot + "/lib/riscv64-linux-gnu"))
+      return "riscv64-linux-gnu";
+    break;
   case llvm::Triple::ppc:
     if (D.getVFS().exists(SysRoot + "/lib/powerpc-linux-gnuspe"))
       return "powerpc-linux-gnuspe";
