From 0b4277953fe2f1c1482552affb6018a9a6121cf5 Mon Sep 17 00:00:00 2001
From: Seppo Yli-Olli <seppo.yliolli@gmail.com>
Date: Fri, 31 Dec 2021 00:03:47 +0200
Subject: [PATCH] Grant us permission to create our symlinks


diff --git a/update-ca-trust b/update-ca-trust
index dd22c0e..233bed6 100644
--- a/update-ca-trust
+++ b/update-ca-trust
@@ -23,6 +23,7 @@ export P11_KIT_NO_USER_CONFIG=1
 # Hashed directory of BEGIN TRUSTED-style certs (usable as OpenSSL CApath and
 # by GnuTLS)
 /usr/bin/p11-kit extract --format=pem-directory-hash --filter=ca-anchors --overwrite --purpose server-auth $DEST/pem/directory-hash
+chmod u+w $DEST/pem/directory-hash
 # Debian compatibility: their /etc/ssl/certs has this bundle
 /usr/bin/ln -s ../tls-ca-bundle.pem $DEST/pem/directory-hash/ca-certificates.crt
 # Backwards compatibility: RHEL/Fedora provided a /etc/ssl/certs/ca-bundle.crt
-- 
2.33.1

