diff --git a/subprojects/gst-plugins-bad/ext/ladspa/meson.build b/subprojects/gst-plugins-bad/ext/ladspa/meson.build
index adfd3459e2..3e8060796d 100644
--- a/subprojects/gst-plugins-bad/ext/ladspa/meson.build
+++ b/subprojects/gst-plugins-bad/ext/ladspa/meson.build
@@ -13,7 +13,7 @@ if get_option('ladspa').disabled()
 endif
 
 # This is an optional dep, but we make it optional only in auto mode
-lrdf_dep = dependency('lrdf', required : get_option('ladspa'))
+lrdf_dep = dependency('lrdf', required: false)
 if lrdf_dep.found()
   ladspa_cargs = ['-DHAVE_LRDF']
 endif
