diff --git a/subprojects/gst-libav/docs/meson.build b/subprojects/gst-libav/docs/meson.build
index 5f25f20837..590b62919a 100644
--- a/subprojects/gst-libav/docs/meson.build
+++ b/subprojects/gst-libav/docs/meson.build
@@ -74,6 +74,6 @@ plugins_doc = [hotdoc.generate_doc('libav',
     gst_c_sources: ['../ext/*/*.[ch]',],
     gst_cache_file: plugins_cache,
     gst_plugin_name: 'libav',
-    dependencies: [gst_dep, gstlibav_plugin, plugins_doc_dep],
+    dependencies: [gst_dep, gstlibav_plugin],
     disable_incremental_build: true,
 )]
