diff --git a/fuzz/build.mk b/fuzz/build.mk
index d38fb90..8e7211d 100644
--- a/fuzz/build.mk
+++ b/fuzz/build.mk
@@ -86,7 +86,7 @@ $(foreach F,$(CORPUS_EXTRA),$(eval \
 
 fuzz/$(OUTPUT_DIR)/fuzz_corpus.stamp: fuzz/original-corpus.tar.gz $(CORPUS_EXTRA)
 	mkdir -p $(CORPUS_DIR)
-	(cd $(CORPUS_DIR); tar xzf ../../original-corpus.tar.gz)
+	(cd $(CORPUS_DIR); tar -x --no-same-owner -zf ../../original-corpus.tar.gz)
 	touch $@
 
 $(foreach B,$(FUZZERS),$(eval \
