From 3cc1b8c138ea26c77d0fce18afb5d0d562331a06 Mon Sep 17 00:00:00 2001
From: Thomas Coldrick <othko97@gmail.com>
Date: Wed, 14 Aug 2019 11:24:19 +0100
Subject: [PATCH] ext/wayland: Define libdrm_dep in meson.build

---
 ext/wayland/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/wayland/meson.build b/ext/wayland/meson.build
index fa6fbf42f..3c377eefb 100644
--- a/ext/wayland/meson.build
+++ b/ext/wayland/meson.build
@@ -8,6 +8,8 @@ wl_sources = [
     'wllinuxdmabuf.c'
 ]
 
+libdrm_dep = dependency('libdrm', version: '>= 2.4.55', required:get_option('wayland'))
+
 if use_wayland
     protocols_datadir = wl_protocol_dep.get_pkgconfig_variable('pkgdatadir')
 
-- 
2.22.0

