diff -ur rpm-4.14.1.old/rpmio/digest_openssl.c rpm-4.14.1/rpmio/digest_openssl.c
--- rpm-4.14.1.old/rpmio/digest_openssl.c	2017-10-05 12:04:57.554602041 +0200
+++ rpm-4.14.1/rpmio/digest_openssl.c	2018-07-19 21:57:17.072663734 +0200
@@ -175,9 +175,6 @@
     case PGPHASHALGO_RIPEMD160:
         return EVP_ripemd160();
 
-    case PGPHASHALGO_MD2:
-        return EVP_md2();
-
     case PGPHASHALGO_SHA256:
         return EVP_sha256();
 
diff -ur rpm-4.14.1.old/rpmio/rpmpgp.c rpm-4.14.1/rpmio/rpmpgp.c
--- rpm-4.14.1.old/rpmio/rpmpgp.c	2017-10-05 12:04:57.569602038 +0200
+++ rpm-4.14.1/rpmio/rpmpgp.c	2018-07-19 21:57:48.624340398 +0200
@@ -89,7 +89,6 @@
     { PGPHASHALGO_MD5,		"MD5" },
     { PGPHASHALGO_SHA1,		"SHA1" },
     { PGPHASHALGO_RIPEMD160,	"RIPEMD160" },
-    { PGPHASHALGO_MD2,		"MD2" },
     { PGPHASHALGO_TIGER192,	"TIGER192" },
     { PGPHASHALGO_HAVAL_5_160,	"HAVAL-5-160" },
     { PGPHASHALGO_SHA256,	"SHA256" },
diff -ur rpm-4.14.1.old/rpmio/rpmpgp.h rpm-4.14.1/rpmio/rpmpgp.h
--- rpm-4.14.1.old/rpmio/rpmpgp.h	2017-10-05 12:04:57.569602038 +0200
+++ rpm-4.14.1/rpmio/rpmpgp.h	2018-07-19 21:57:40.336425316 +0200
@@ -259,7 +259,6 @@
     PGPHASHALGO_MD5		=  1,	/*!< MD5 */
     PGPHASHALGO_SHA1		=  2,	/*!< SHA1 */
     PGPHASHALGO_RIPEMD160	=  3,	/*!< RIPEMD160 */
-    PGPHASHALGO_MD2		=  5,	/*!< MD2 */
     PGPHASHALGO_TIGER192	=  6,	/*!< TIGER192 */
     PGPHASHALGO_HAVAL_5_160	=  7,	/*!< HAVAL-5-160 */
     PGPHASHALGO_SHA256		=  8,	/*!< SHA256 */
diff -ur rpm-4.14.1.old/sign/rpmsignfiles.c rpm-4.14.1/sign/rpmsignfiles.c
--- rpm-4.14.1.old/sign/rpmsignfiles.c	2017-10-05 12:04:57.588602034 +0200
+++ rpm-4.14.1/sign/rpmsignfiles.c	2018-07-19 21:58:03.104192063 +0200
@@ -21,7 +21,6 @@
     [PGPHASHALGO_MD5]          = "md5",
     [PGPHASHALGO_SHA1]         = "sha1",
     [PGPHASHALGO_RIPEMD160]    = "rmd160",
-    [PGPHASHALGO_MD2]          = "md2",
     [PGPHASHALGO_TIGER192]     = "tgr192",
     [PGPHASHALGO_HAVAL_5_160]  = "haval5160",
     [PGPHASHALGO_SHA256]       = "sha256",
