diff --git a/dracut.sh b/dracut.sh
index bfb7a1ac..9bab25c8 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1331,6 +1331,9 @@ if [[ ! $print_cmdline ]]; then
             i?86)
                 EFI_MACHINE_TYPE_NAME=ia32
                 ;;
+            riscv64)
+                EFI_MACHINE_TYPE_NAME=riscv64
+                ;;
             *)
                 dfatal "Architecture '$(uname -m)' not supported to create a UEFI executable"
                 exit 1
