diff --git a/lib/efi_loader/efi_device_path.c b/lib/efi_loader/efi_device_path.c
index 398dbc699b..3ec126c9f4 100644
--- a/lib/efi_loader/efi_device_path.c
+++ b/lib/efi_loader/efi_device_path.c
@@ -784,15 +784,11 @@ static void *dp_part_node(void *buf, struct blk_desc *desc, int part)
 			break;
 		case SIG_TYPE_MBR:
 			hddp->signature_type = 1;
-			memset(hddp->partition_signature, 0,
-			       sizeof(hddp->partition_signature));
-			memcpy(hddp->partition_signature, &desc->mbr_sig,
-			       sizeof(desc->mbr_sig));
+                        uuid_str_to_bin(info.uuid, hddp->partition_signature, 1);
 			break;
 		case SIG_TYPE_GUID:
 			hddp->signature_type = 2;
-			memcpy(hddp->partition_signature, &desc->guid_sig,
-			       sizeof(hddp->partition_signature));
+                        uuid_str_to_bin(info.uuid, hddp->partition_signature, 1);
 			break;
 		}
 
