diff --git a/autogen.sh b/autogen.sh
index 7a0783623..97a0e07ee 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -29,15 +29,15 @@ rm -rf autom4te.cache
 
 echo "Generating configuration files for $package, please wait...."
 
-echo "  running autoreconf"
-autoreconf || exit 1
-
 if test ! -x config.guess -o ! -x config.sub ; then
   rm -f config.guess config.sub
   cp `automake --print-libdir`/config.guess . || exit 1
   cp `automake --print-libdir`/config.sub . || exit 1
 fi
 
+echo "  running autoreconf"
+autoreconf || exit 1
+
 if test ! -x install-sh ; then
   rm -f install-sh
   cp `automake --print-libdir`/install-sh . || exit 1
