diff --git a/src/modules/meson.build b/src/modules/meson.build
index 3e0e3a3fd..0f773425c 100644
--- a/src/modules/meson.build
+++ b/src/modules/meson.build
@@ -42,7 +42,6 @@ pipewire_module_access = shared_library('pipewire-module-access', [ 'module-acce
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : pipewire_module_access_deps
 )
 
@@ -51,7 +50,6 @@ pipewire_module_loopback = shared_library('pipewire-module-loopback',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -111,7 +109,6 @@ pipewire_module_filter_chain = shared_library('pipewire-module-filter-chain',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   link_with : simd_dependencies,
   dependencies : filter_chain_dependencies,
 )
@@ -125,7 +122,6 @@ pipewire_module_echo_cancel = shared_library('pipewire-module-echo-cancel',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, pipewire_dep],
 )
 
@@ -135,7 +131,6 @@ pipewire_module_profiler = shared_library('pipewire-module-profiler',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -143,7 +138,6 @@ pipewire_module_rt = shared_library('pipewire-module-rt', [ 'module-rt.c' ],
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [dbus_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -155,7 +149,6 @@ pipewire_module_rtkit = shared_library('pipewire-module-rtkit', [ 'module-rt.c'
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [dbus_dep, mathlib, dl_lib, pipewire_dep],
 )
 endif
@@ -167,7 +160,6 @@ pipewire_module_portal = shared_library('pipewire-module-portal', [ 'module-port
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [dbus_dep, mathlib, dl_lib, pipewire_dep],
 )
 endif
@@ -181,7 +173,6 @@ pipewire_module_client_device = shared_library('pipewire-module-client-device',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -190,7 +181,6 @@ pipewire_module_link_factory = shared_library('pipewire-module-link-factory',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -211,7 +201,6 @@ pipewire_module_protocol_native = shared_library('pipewire-module-protocol-nativ
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : pipewire_module_protocol_deps,
 )
 
@@ -291,7 +280,6 @@ pipewire_module_protocol_pulse = shared_library('pipewire-module-protocol-pulse'
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : pipewire_module_protocol_pulse_deps,
 )
 
@@ -303,7 +291,6 @@ if build_module_pulse_tunnel
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, pipewire_dep, pulseaudio_dep],
 )
 endif
@@ -314,7 +301,6 @@ pipewire_module_pipe_tunnel = shared_library('pipewire-module-pipe-tunnel',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, pipewire_dep],
 )
 
@@ -323,7 +309,6 @@ pipewire_module_protocol_simple = shared_library('pipewire-module-protocol-simpl
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : pipewire_module_protocol_deps,
 )
 
@@ -332,7 +317,6 @@ pipewire_module_example_sink = shared_library('pipewire-module-example-sink',
   include_directories : [configinc],
   install : false,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -341,7 +325,6 @@ pipewire_module_example_sink = shared_library('pipewire-module-example-source',
   include_directories : [configinc],
   install : false,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -358,7 +341,6 @@ pipewire_module_client_node = shared_library('pipewire-module-client-node',
   link_with : pipewire_module_protocol_native,
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -371,7 +353,6 @@ pipewire_module_metadata = shared_library('pipewire-module-metadata',
   link_with : pipewire_module_protocol_native,
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -410,7 +391,6 @@ pipewire_module_adapter = shared_library('pipewire-module-adapter',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep],
 )
 
@@ -432,7 +412,6 @@ pipewire_module_session_manager = shared_library('pipewire-module-session-manage
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [spa_dep, mathlib, dl_lib, pipewire_dep],
 )
 
@@ -445,7 +424,6 @@ pipewire_module_zeroconf_discover = shared_library('pipewire-module-zeroconf-dis
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, avahi_dep],
 )
 endif
@@ -459,7 +437,6 @@ pipewire_module_raop_discover = shared_library('pipewire-module-raop-discover',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, avahi_dep],
 )
 endif
@@ -473,7 +450,6 @@ pipewire_module_raop_sink = shared_library('pipewire-module-raop-sink',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, openssl_lib],
 )
 endif
@@ -489,7 +465,6 @@ pipewire_module_roc_sink = shared_library('pipewire-module-roc-sink',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, roc_lib],
 )
 
@@ -498,7 +473,6 @@ pipewire_module_roc_source = shared_library('pipewire-module-roc-source',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, roc_lib],
 )
 endif
@@ -512,7 +486,6 @@ pipewire_module_x11_bell = shared_library('pipewire-module-x11-bell',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep, x11_dep, xfixes_dep, canberra_dep],
 )
 endif
@@ -523,7 +496,6 @@ pipewire_module_fallback_sink = shared_library('pipewire-module-fallback-sink',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep],
 )
 
@@ -548,7 +520,6 @@ pipewire_module_avb = shared_library('pipewire-module-avb',
   include_directories : [configinc],
   install : true,
   install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
   dependencies : [mathlib, dl_lib, rt_lib, pipewire_dep],
 )
 endif
