diff -u -r elfutils.old/libdw/Makefile.am elfutils/libdw/Makefile.am
--- elfutils.old/libdw/Makefile.am	2019-02-20 12:54:32.024228734 +0100
+++ elfutils/libdw/Makefile.am	2019-02-20 12:57:55.978307676 +0100
@@ -115,7 +115,7 @@
 # not fly in a setuid executable that links in libdw.
 	$(AM_V_CCLD)$(LINK) $(dso_LDFLAGS) -o $@ \
 		-Wl,--soname,$@.$(VERSION) \
-		-Wl,--enable-new-dtags,-rpath,$(pkglibdir) \
+		-Wl,--enable-new-dtags \
 		-Wl,--version-script,$<,--no-undefined \
 		-Wl,--whole-archive $(libdw_so_LIBS) -Wl,--no-whole-archive \
 		$(libdw_so_LDLIBS)
diff -u -r elfutils.old/libebl/eblopenbackend.c elfutils/libebl/eblopenbackend.c
--- elfutils.old/libebl/eblopenbackend.c	2019-02-20 12:54:32.040228584 +0100
+++ elfutils/libebl/eblopenbackend.c	2019-02-20 12:54:38.508167731 +0100
@@ -316,7 +316,7 @@
    It might also work for shared libraries when installed in
    ${prefix}/lib/ or ${prefix}/lib64/, but not for multiarch
    library installs like ${prefix}/lib/i386-linux-gnu/  */
-#define BINORIGINDIR "$ORIGIN/../$LIB/" LIBEBL_SUBDIR "/"
+#define BINORIGINDIR "$ORIGIN/" LIBEBL_SUBDIR "/"
 
 /* This works if libebl has been linked into a shared library,
    just look in the subdir.  */
