diff --git a/src/shared/bootspec.c b/src/shared/bootspec.c
index 13d7b2f160..7a9ebf086f 100644
--- a/src/shared/bootspec.c
+++ b/src/shared/bootspec.c
@@ -1032,6 +1032,9 @@ static int verify_fsroot_dir(
                                       (unprivileged_mode && errno == EACCES) ? LOG_DEBUG : LOG_ERR, errno,
                                       "Failed to determine block device node of \"%s\": %m", path);
 
+        if (getenv_bool("SYSTEMD_RELAX_ESP_CHECKS") > 0)
+                goto out;
+
         if (major(st.st_dev) == 0)
                 return log_full_errno(searching ? LOG_DEBUG : LOG_ERR,
                                       SYNTHETIC_ERRNO(searching ? EADDRNOTAVAIL : ENODEV),
@@ -1068,6 +1071,7 @@ static int verify_fsroot_dir(
                                       SYNTHETIC_ERRNO(searching ? EADDRNOTAVAIL : ENODEV),
                                       "Directory \"%s\" is not the root of the file system.", path);
 
+ out:
         if (ret_dev)
                 *ret_dev = st.st_dev;
 
