From 2b74963945687a4e57b5a1214bcae693efbd6126 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Javier=20Jard=C3=B3n?= <jjardon@gnome.org>
Date: Sun, 6 Dec 2020 17:39:11 +0000
Subject: [PATCH] Add support for RISC-V

---
 webrtc/base/basictypes.h | 4 ++++
 webrtc/typedefs.h        | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/webrtc/base/basictypes.h b/webrtc/base/basictypes.h
index 4c3d5d1..ce0fe11 100644
--- a/webrtc/base/basictypes.h
+++ b/webrtc/base/basictypes.h
@@ -29,6 +29,10 @@
 #define CPU_ARM 1
 #endif
 
+#if defined(__riscv) || defined(_M_RISCV)
+#define CPU_RISCV 1
+#endif
+
 #if defined(CPU_X86) && defined(CPU_ARM)
 #error CPU_X86 and CPU_ARM both defined.
 #endif
diff --git a/webrtc/typedefs.h b/webrtc/typedefs.h
index d875490..657e748 100644
--- a/webrtc/typedefs.h
+++ b/webrtc/typedefs.h
@@ -44,6 +44,13 @@
 #elif defined(__MIPSEL__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__riscv)
+#if __riscv_xlen == 64
+# define WEBRTC_ARCH_64_BITS
+#else
+# define WEBRTC_ARCH_32_BITS
+#endif
+#define WEBRTC_ARCH_LITTLE_ENDIAN
 #elif defined(__pnacl__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
-- 
2.28.0

