diff --git a/sdl2-config.cmake.in b/sdl2-config.cmake.in
index c570511..08c2a89 100644
--- a/sdl2-config.cmake.in
+++ b/sdl2-config.cmake.in
@@ -71,11 +71,4 @@ else() # (not WIN32) or MSVC
 
 endif() # (not WIN32) or MSVC
 
-  add_library(SDL2::SDL2-static STATIC IMPORTED)
-  set_target_properties(SDL2::SDL2-static PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${SDL2_INCLUDE_DIRS}"
-    IMPORTED_LINK_INTERFACE_LANGUAGES "C"
-    IMPORTED_LOCATION "${SDL2_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}SDL2${CMAKE_STATIC_LIBRARY_SUFFIX}"
-    INTERFACE_LINK_LIBRARIES "${SDL2_EXTRA_LINK_FLAGS_STATIC}")
-
 endif() # NOT TARGET SDL2::SDL2
