commit 704c5bdaca4be3b61b4df38087ec9bc17f9aa2d6
Author: Akira TAGOH <akira@tagoh.org>
Date:   Mon Jun 10 10:57:05 2019 +0000

    Fix endianness on generating MD5 cache name

diff --git a/src/fccache.c b/src/fccache.c
index 3071104..e027f7f 100644
--- a/src/fccache.c
+++ b/src/fccache.c
@@ -1637,7 +1637,7 @@ FcCacheNumFont args1(const FcCache *c)
  * will fill a supplied 16-byte array with the digest.
  */
 
-#ifndef HIGHFIRST
+#ifndef WORDS_BIGENDIAN
 #define byteReverse(buf, len)	/* Nothing */
 #else
 /*
