commit 01d3e2abe08074316eea1449d70eb9fe9f39d493
Author: Akira TAGOH <akira@tagoh.org>
Date:   Wed Aug 28 14:04:21 2019 +0900

    Drop doc to get make check successul.

diff --git a/doc/fcstring.fncs b/doc/fcstring.fncs
index d5ec043..0412bbd 100644
--- a/doc/fcstring.fncs
+++ b/doc/fcstring.fncs
@@ -223,18 +223,6 @@ This is just a wrapper around free(3) which helps track memory usage of
 strings within the fontconfig library.
 @@
 
-@RET@		FcChar8 *
-@FUNC@		FcStrBuildFilename
-@TYPE1@		const FcChar8 *			@ARG1@		path
-@TYPE2@		...
-@PURPOSE@	Concatenate strings as a file path
-@DESC@
-Creates a filename from the given elements of strings as file paths
-and concatenate them with the appropriate file separator.
-Arguments must be null-terminated.
-This returns a newly-allocated memory which should be freed when no longer needed.
-@@
-
 @RET@		FcChar8 *
 @FUNC@		FcStrDirname
 @TYPE1@		const FcChar8 *			@ARG1@		file
