commit a901f941ae6b985aef1d358d3a180530764ec65b
Author: Akira TAGOH <akira@tagoh.org>
Date:   Wed Aug 28 13:44:12 2019 +0900

    Drop new symbols and disable test cases

diff --git a/fontconfig/fontconfig.h b/fontconfig/fontconfig.h
index af870d0..bac1dda 100644
--- a/fontconfig/fontconfig.h
+++ b/fontconfig/fontconfig.h
@@ -1076,10 +1076,6 @@ FcUtf16Len (const FcChar8   *string,
 	    int		    *nchar,
 	    int		    *wchar);
 
-FcPublic FcChar8 *
-FcStrBuildFilename (const FcChar8 *path,
-		    ...);
-
 FcPublic FcChar8 *
 FcStrDirname (const FcChar8 *file);
 
diff --git a/src/fcint.h b/src/fcint.h
index 9ae00be..3f30b3e 100644
--- a/src/fcint.h
+++ b/src/fcint.h
@@ -1313,6 +1313,10 @@ FcStrUsesHome (const FcChar8 *s);
 FcPrivate FcBool
 FcStrIsAbsoluteFilename (const FcChar8 *s);
 
+FcPrivate FcChar8 *
+FcStrBuildFilename (const FcChar8 *path,
+		    ...);
+
 FcPrivate FcChar8 *
 FcStrLastSlash (const FcChar8  *path);
 
diff --git a/test/Makefile.am b/test/Makefile.am
index 3068d69..e714363 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -75,24 +75,24 @@ check_PROGRAMS += test-bz106618
 test_bz106618_LDADD = $(top_builddir)/src/libfontconfig.la
 
 if !OS_WIN32
-check_PROGRAMS += test-bz106632
-test_bz106632_CFLAGS =					\
-	-I$(top_builddir)				\
-	-I$(top_builddir)/src				\
-	-I$(top_srcdir)					\
-	-I$(top_srcdir)/src				\
-	-DFONTFILE='"$(abs_top_srcdir)/test/4x6.pcf"'	\
-	-DHAVE_CONFIG_H					\
-	$(NULL)
-test_bz106632_LDADD = $(top_builddir)/src/libfontconfig.la
-TESTS += test-bz106632
+#check_PROGRAMS += test-bz106632
+#test_bz106632_CFLAGS =					\
+#	-I$(top_builddir)				\
+#	-I$(top_builddir)/src				\
+#	-I$(top_srcdir)					\
+#	-I$(top_srcdir)/src				\
+#	-DFONTFILE='"$(abs_top_srcdir)/test/4x6.pcf"'	\
+#	-DHAVE_CONFIG_H					\
+#	$(NULL)
+#test_bz106632_LDADD = $(top_builddir)/src/libfontconfig.la
+#TESTS += test-bz106632
 endif
 
-check_PROGRAMS += test-issue107
-test_issue107_LDADD =					\
-	$(top_builddir)/src/libfontconfig.la		\
-	$(NULL)
-TESTS += test-issue107
+#check_PROGRAMS += test-issue107
+#test_issue107_LDADD =					\
+#	$(top_builddir)/src/libfontconfig.la		\
+#	$(NULL)
+#TESTS += test-issue107
 
 EXTRA_DIST=run-test.sh run-test-conf.sh $(TESTDATA) out.expected-long-family-names out.expected-no-long-family-names
 
