From 699c5a573eddccebd9ad938cb79900547a29d2a5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <b@bpiotrowski.pl>
Date: Thu, 30 Apr 2020 15:29:55 +0200
Subject: [PATCH] Allow timestamp in the future in validate-relax

---
 libappstream-glib/as-app-validate.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libappstream-glib/as-app-validate.c b/libappstream-glib/as-app-validate.c
index 742b968..f60c18a 100644
--- a/libappstream-glib/as-app-validate.c
+++ b/libappstream-glib/as-app-validate.c
@@ -815,11 +815,13 @@ as_app_validate_release (AsApp *app,
 	guint number_para_max = 3;
 	guint number_para_min = 1;
 	gboolean required_timestamp = TRUE;
+	gboolean required_past_timestamp = TRUE;
 
 	/* relax the requirements a bit */
 	if ((helper->flags & AS_APP_VALIDATE_FLAG_RELAX) > 0) {
 		number_para_max = 4;
 		required_timestamp = FALSE;
+		required_past_timestamp = FALSE;
 	}
 
 	/* check version */
@@ -844,7 +846,7 @@ as_app_validate_release (AsApp *app,
 	}
 
 	/* check the timestamp is not in the future */
-	if (timestamp > (guint64) g_get_real_time () / G_USEC_PER_SEC) {
+	if (required_past_timestamp && timestamp > (guint64) g_get_real_time () / G_USEC_PER_SEC) {
 		ai_app_validate_add (helper,
 				     AS_PROBLEM_KIND_ATTRIBUTE_INVALID,
 				     "<release> timestamp is in the future");
-- 
2.26.2

