From 9406791836e914fd64351b21b035d0b6e933324e Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tmraz@fedoraproject.org>
Date: Mon, 26 Oct 2020 15:23:24 +0100
Subject: [PATCH] Use proper record version in client hello after hello retry
 request

Signed-off-by: Tomas Mraz <tmraz@fedoraproject.org>

Fixes: #1053
---
 lib/record.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/record.c b/lib/record.c
index af993fe6e..faf53f562 100644
--- a/lib/record.c
+++ b/lib/record.c
@@ -367,6 +367,7 @@ copy_record_version(gnutls_session_t session,
 	lver = get_version(session);
 	if (session->internals.initial_negotiation_completed ||
 	    htype != GNUTLS_HANDSHAKE_CLIENT_HELLO ||
+	    (session->internals.hsk_flags & HSK_HRR_RECEIVED) ||
 	    session->internals.default_record_version[0] == 0) {
 
 		if (unlikely(lver == NULL))
-- 
2.33.1
