diff -ur rpm-4.14.1.old/configure.ac rpm-4.14.1/configure.ac
--- rpm-4.14.1.old/configure.ac	2018-01-16 10:30:14.074141024 +0100
+++ rpm-4.14.1/configure.ac	2018-07-19 19:11:08.117863176 +0200
@@ -519,20 +519,6 @@
 
 case "$with_external_db" in
 yes )
-  AC_CHECK_HEADERS([db.h],[
-    AC_PREPROC_IFELSE([
-      AC_LANG_SOURCE([
-	#include <db.h>
-	#if ((DB_VERSION_MAJOR < 4) || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR < 5))
-	#error Berkeley DB too old
-	#endif
-      ])
-    ],[ WITH_DB_LIB=-ldb ], 
-      [ AC_MSG_ERROR([Berkeley DB version >= 4.5 required])
-    ])
-  ],[
-    AC_MSG_ERROR([missing required header db.h])
-  ])
   ;;
 no|maybe )
   # Try internal database first, then fall back to external
@@ -1052,10 +1038,6 @@
 
 AC_SUBST(OBJDUMP)
 
-if test "$with_external_db" = no; then
-    AC_CONFIG_SUBDIRS(db3)
-fi
-
 AM_CONDITIONAL([WITH_INTERNAL_DB],[test "$with_external_db" = no])
 AM_CONDITIONAL([DOXYGEN],[test "$DOXYGEN" != no])
 AM_CONDITIONAL([HACKINGDOCS],[test "$with_hackingdocs" = yes])
